database/interface.py

```diff
@@ -232,7 +232,11 @@ def add_harvest_records(self, records_data: list) -> bool:
         :raises Exception: if the records_data contains records with errors
         """
         try:
-            self.db.bulk_insert_mappings(HarvestRecord, records_data)
+            for i, record_data in enumerate(records_data):
+                new_record = HarvestRecord(**record_data)
+                self.db.add(new_record)
+                if i % 1000 == 0:
+                    self.db.flush()
```

Comment:
I don't fully understand how this is better. I thought we wanted to track the ID in code, or insert individually into the DB and get back the ID such that we didn't have to make a lookup call later.
This seems to not do the bulk insert, but I don't see how it adds/tracks the generated ID through the rest of the pipeline.

Reply:
you could update the return to a dict like { <identifier>: <new_record.id> } then pass it along to ckan

Reply:
Glad I had to update the branch and we got this comment, James! I was solely focused on the issue of the foreign keys, and totally forgot that we wanted to export the id's for use in CKAN.

What is best form for that, @rshewitt? 

- Attach `id` to `record_data` and return the list of dicts
- Cast newly created HarvestRecords to a list of dicts
- A single dict in the form you mentioned above?

All are easy enough.

Reply:
i think returning a dictionary would be more efficient than a list in this circumstance. using a list would require a loop to find the right record but a dict with the format of `{ <dataset id>: <new_record.id> }` you can derive the new_record.id instantly.

Reply:
Sounds good. The only thing I don't like is our `add_harvest_record` method returns a different signature, but maybe we can think more about that as a simple lookup table is definitely the right way to go in this case.